File: scala/src/test/scala/LifeSpec.scala
import org.scalatest.{FunSpec, Matchers}

class LifeSpec extends FunSpec with Matchers {
  describe("Initialisation") {
    it("creates a set with the correct number of cells") {
      val setOfCells = Set(Cell(1,1), Cell(2 ,2))
      val life = new Life(setOfCells)

      life.getLiveCells.size shouldEqual 2
    }
  }

  describe("Underpopulation") {
    it("?????") {
      val setOfCells = Set(Cell(1,1), Cell(2 ,2))
      val life = new Life(setOfCells)

      life.cellShouldSurvive(0) shouldBe false
      life.cellShouldSurvive(1) shouldBe false
    }
  }
}

File: ruby/spec/life_spec.rb
require 'spec_helper'

include GameOfLife

describe Life do
  
  it "can be initialized with an array of cells" do
    initial_state = [Cell.new(1, 1), Cell.new(2, 2)]
    life = Life.new(initial_state)
    expect(initial_state).to eq(life.live_cells.to_a)
  end

  it "can test underpopulation" do
    expect(Life.cell_should_survive(0)).to be false
    expect(Life.cell_should_survive(1)).to be false
  end
end
